### PR TITLE
CI: Set pytest minversion to 4.0.2

### LINKED ISSUE #23519: pytest 3.10 minversion
Bug in parsing minversion: https://github.com/pytest-dev/pytest/issues/4306

this issue is to restore it.

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [x] closes #23519
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
would need to make a min on the install scripts as well (when we have a working version)

**Comment by TomAugspurger** [comment]
You're saying pin a minimum version of pytest for CI? What would the
benefit of that be? Do we care about
supporting old versions of pytest?

**Comment by simonjayhawkins** [comment]
currently using pytest-3.8.1 on macOS py35_np_120

should we update the ci build or downgrade the pytest minversion?

**Comment by simonjayhawkins** [comment]
the minversion should probably be above pytest 3.10 otherwise may encounter the minversion parse error xref #23519 again in the future.

File: ci/deps/azure-macos-35.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ dependencies:
   - xlsxwriter
   - xlwt
   # universal
-  - pytest
+  - pytest>=4.0.2
```
Comment (jreback) [inline ci/deps/azure-macos-35.yaml]: i would add it to all files just to indicate we have a min version explicitly

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/deps/azure-macos-35.yaml (hunk 1 above)
Comment (simonjayhawkins) [reply ci/deps/azure-macos-35.yaml]: OK, if this succeeds, i'll do that.

**Comment by jreback** [comment]
also pls update the docs (i think we have a reference to pytest and it’s min version iirc);

also would need a PR against both MacPython/pandas-wheels and conda-forge/pandas-feedstock

when this is merged

**Comment by simonjayhawkins** [comment]
it appears that pytest-3.8.1 is the latest version for py35 on conda

**Comment by jreback** [comment]
you can try moving to the pip section

**Comment by simonjayhawkins** [comment]
> i think we have a reference to pytest and it’s min version iirc

that's right. contributing.rst and install.rst will need updating.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. @pandas-dev/pandas-core any objections?

**Review by gfyoung — APPROVED** [review]

File: doc/source/whatsnew/v0.25.0.rst [hunk 2]
```diff
@@ -31,6 +31,20 @@ Backwards incompatible API changes
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We have updated our minimum supported versions of dependencies (:issue:`23519`).
+If installed, we now require:
+
++-----------------+-----------------+----------+
+| Package         | Minimum Version | Required |
++=================+=================+==========+
+| pytest (dev)    | 4.0.2           |          |
++-----------------+-----------------+----------+
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.25.0.rst]: I don't think we should put that here, that is only for contributors, not for pandas users?

**Review by jorisvandenbossche — COMMENTED** [review]
Apart from the whatsnew comment, fine for me.

File: doc/source/whatsnew/v0.25.0.rst (hunk 2 above)
Comment (gfyoung) [reply doc/source/whatsnew/v0.25.0.rst]: @jorisvandenbossche : This was already done in `0.24.0`, though I could go either way on this.

Comment (TomAugspurger) [reply doc/source/whatsnew/v0.25.0.rst]: This is kinda exposed to users via `pd.test`, so let's keep.

**Comment by TomAugspurger** [comment]
Thanks @simonjayhawkins!

**Pull request merged**